pkg/plugin/nvidia/server.go

```diff
@@ -299,14 +353,21 @@ func (m *NvidiaDevicePlugin) Allocate(ctx context.Context, reqs *pluginapi.Alloc
 	var candidatePod *v1.Pod
 	for _, pod := range availablePods {
 		for i, c := range pod.Spec.Containers {
-			if !IsGPURequiredContainer(&c) {
+			if !IsGPURequiredContainer(&c, m.resourceName) {
 				continue
 			}
-
-			if GetGPUResourceOfContainer(&pod.Spec.Containers[i]) == firstContainerReqDeviceCount {
-				klog.Infof("Got candidate Pod %s(%s), the device count is: %d", pod.UID, c.Name, firstContainerReqDeviceCount)
-				candidatePod = pod
-				goto Allocate
+			if m.resourceName == VolcanoGPUMemory {
+				if GetGPUResourceOfContainer(&pod.Spec.Containers[i], VolcanoGPUMemory) == firstContainerReqDeviceCount {
```

Comment:
@peiniliu Actually i don't know exactaly the calculation of `firstContainerReqDeviceCount`. Can you explain more about this?
	firstContainerReq := reqs.ContainerRequests[0]
	firstContainerReqDeviceCount := uint(len(firstContainerReq.DevicesIDs))

Reply:
I did not change this compared to the previous version. I think your question point to this issue #25

Reply:
kubelet will launch `Allocate` for each container, though the argument is containers list, it only contains one container. That's why `firstContainerReq := reqs.ContainerRequests[0]`

Reply:
maybe it's better to add some comments here to clarify this background.

Reply:
added comments here.

Reply:
> kubelet will launch `Allocate` for each container, though the argument is containers list, it only contains one container. That's why `firstContainerReq := reqs.ContainerRequests[0]`
@peiniliu @shinytang6 
I know currently we only support the first container case.  I want to understand the meaning of this line that peiniliu added.  I thought firstContainerReqDeviceCount stands for device num of first container of pod.  How we can compare it with VolcanoGPUMemory amount?
`GetGPUResourceOfContainer(&pod.Spec.Containers[i], VolcanoGPUMemory) == firstContainerReqDeviceCount`

Reply:
Hi,
In shared mode, devices are virtual devices. The total devices the kubelet managed equals to totalMemory/memory unit.
so the VolcanoGPUMemory = virtual devices count